src/util.rs

```diff
@@ -209,7 +229,7 @@ pub(crate) mod ptr {
         }
     }
 
-    impl<'a> Ptr<'a, [u8]> {
+    impl<'a> Ptr<'a, [u8], true> {
```

Comment:
I think this PR is just about ready for review. The last blocker to supporting unaligned types is removing the alignment requirements from the items in this `impl` block.